libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go

```diff
@@ -609,18 +619,89 @@ func (dhtPeer *DHTPeer) Close() []error {
 	return status
 }
 
+func generate_x509_cert() (*tls.Certificate, error) {
+	privBtcKey, err := btcec.NewPrivateKey(elliptic.P256())
```

Comment:
Please provide a docstring, in particular it needs to explain that we use a session key and for what etc.

Reply:
done